Open Judge System/Services/OJS.Services/Data/Participants/ParticipantsDataService.cs

```diff
@@ -139,6 +144,36 @@ public void Delete(IEnumerable<Participant> participantsForDeletion)
                     IsInvalidated = true
                 });
 
+        public void UpdateTotalScoreSnapshot()
+        {
+            var command = @"DECLARE @BatchSize INT = 5000;
+               DECLARE @MaxId INT = (SELECT MAX(Id) FROM Participants);
+               DECLARE @Offset INT = 0;
+
+               WHILE @Offset <= @MaxId
+               BEGIN
+                   WITH OrderedParticipants AS (
+                   SELECT TOP (@BatchSize) Id
+               FROM Participants
+               WHERE Id > @Offset
+               ORDER BY Id
+                   )
+               UPDATE p
+               SET TotalScoreSnapshot = ISNULL((
+                   SELECT SUM(ps.Points)
+               FROM ParticipantScores ps
+                   JOIN Problems pr ON ps.ProblemId = pr.Id AND pr.IsDeleted = 0
+               WHERE ps.ParticipantId = p.Id
+                   ), 0)
+               FROM OrderedParticipants op
+                   JOIN Participants p ON op.Id = p.Id;
+
+               SET @Offset = @Offset + @BatchSize;
+               END;";
```

Comment:
It is good idea to extract variables for `ps.Points`, `ps.ProblemId` and the rest with `nameof()`. 
This way we will have compile time notion in the code that these properties are used in this query. If changed, the person working on it will see it, as now this can be easily missed and broken.

Reply:
Done